critiquebrainz/frontend/views/review.py

```diff
@@ -125,13 +119,14 @@ def revisions(id):
         raise NotFound("Can't find a review with the specified ID.")
     if review.is_hidden and not current_user.is_admin():
         raise NotFound(gettext("Review has been hidden."))
-
-    revisions = Revision.query.filter_by(review=review)
-    count = revisions.count()
-    revisions = revisions.order_by(desc(Revision.timestamp)).limit(RESULTS_LIMIT)
+    try:
+        revisions, count = db_revision.get(id)
+    except db_exceptions.NoDataFoundException:
+        raise NotFound(gettext("The revision(s) you are looking for does not exist."))
+    votes = db_revision.get_votes(id)
+    revisions = revisions[:RESULTS_LIMIT]
```

Comment:
The problem here is that you are retrieving all revisions of a review just to throw them away. One way to improve this is to allow passing `limit` and `offset` arguments into the `get` function.